### DISCUSSION TITLE
GitHub Access private file by URL

### DISCUSSION THREAD

**Discussion by vbajrhd** [community/community#53637] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have one file in private repository in github.com. I generate a fine-grained token with access Read-only for:"Pull requests" and "Metadata(mandatory)".

I tried download file by serveral URLs using:

`https://myuser:token_generated@github.com/myuser/myrepository/file.txt` give me a blank page with only message "Not Found"

`https://myuser:token_generated@github.com/myuser/myrepository/blob/file.txt` `https://myuser:token_generated@github.com/myuser/myrepository/blob/main/file.txt` Send me to a personal page github 404 message "This is not the web page you are looking for."

`https://myuser:token_generated@raw.githubusercontent.com/myuser/myrepository/main/file.txt` give me a blank page with message "404:not Found"

I tryed this from a answer in comunity: `https://github.com/MYUSER/MYREPOSITORY/blob/main/file.txt -H “Authorization: Token $MYTOKEN”`  - Send me to a personal page github 404 message "This is not the web page you are looking for."

How can i write correct URL to download, or i need to give more access to Token?

Thanks all in advance!!!

**Comment by airtower-luna** [comment]
The last approach with the `Authorization` header is correct in principle. Just try to use the [Get repository content API endpoint](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#get-repository-content) instead of the URL you'd use in the web UI, like so:

```
https://api.github.com/repos/USER/REPO/contents/file.txt?ref=main
```

Mind that by default that'll return JSON that contains information about the file, URLs to find/download it (including a URL for raw data), and for small files the data in a `content` field (usually base64 encoded). To get just the raw file, add an `Accept` header like this:

```
Accept: application/vnd.github.raw
```

Using the `https://raw.githubusercontent.com/...` URL with `Authorization` header should work, too, but I haven't seen it documented as API, so I wouldn't rely on it too much. :sweat_smile:

**Reply by vbajrhd** [reply]
Hi airtower-luna, thank you for your answer! 
I try `https://api.github.com/repos/MYUSER/MYREPO/contents/file.txt?ref=main`
and i get:
{
  "message": "Not Found",
  "documentation_url": `"https://docs.github.com/rest/reference/repos#get-repository-content"`
}
not the json file. Am i doing something wrong? PS my repo only have one file=file.txt and is private.

**Reply by airtower-luna** [reply]
Did you send the `Authorization` header? The token needs to have the **repo** scope.
```
Authorization: Bearer <YOUR-TOKEN>
```

**Reply by vbajrhd** [reply]
Thanks for your help! I think I don't have enough knowledge for this. I will give up. Thanks!

**Reply by airtower-luna** [reply]
It's really not complicated, how to set it just depends on which HTTP client (or library) you use. You had the `-H "Authorization: Bearer <YOUR-TOKEN>"` style curl uses above, that's exactly right if the token is valid and has the right scope (you set that when generating the token).
